crates/biome_css_formatter/src/utils/component_value_list.rs

```diff
@@ -216,3 +236,35 @@ where
         ValueListLayout::Fill
     }
 }
+
+pub(crate) fn use_one_group_per_line<N, I>(css_property: Option<String>, list: &N) -> bool
+where
+    N: AstNodeList<Language = CssLanguage, Node = I> + AstNode<Language = CssLanguage>,
+    I: AstNode<Language = CssLanguage> + IntoFormat<CssFormatContext>,
+{
+    let is_css_property = css_property.is_some();
+    let is_custom_property = css_property.is_some_and(|name| name.starts_with("--"));
+    if !is_css_property || is_custom_property {
+        return false;
+    }
+
+    let mut group_count = 0;
+    let mut group_size = 0;
+    let mut max_group_size = 0;
+    for item in list.iter() {
+        let token_kind = CssGenericDelimiter::cast_ref(item.syntax())
+            .and_then(|node| node.value().ok())
+            .map(|token| token.kind());
+        if matches!(token_kind, Some(CssSyntaxKind::COMMA)) {
+            group_count += 1;
+            max_group_size = cmp::max(group_size, max_group_size);
+            group_size = 0;
+            continue;
+        }
+        group_size += 1;
+    }
+    group_count += 1;
+    max_group_size = cmp::max(group_size, max_group_size);
```

Comment:
Why do we calculate `max_group_size` *again*? You might want to consider adding a comment explaining why there's this logic because it's unclear.